pkg/operator/observe_config.go

```diff
@@ -90,69 +89,114 @@ func (c ConfigObserver) sync() error {
 	// don't worry about errors
 	currentConfig := map[string]interface{}{}
 	json.NewDecoder(bytes.NewBuffer(operatorConfig.Spec.ObservedConfig.Raw)).Decode(&currentConfig)
-	if reflect.DeepEqual(currentConfig, observedConfig) {
-		return nil
+
```

Comment:
you've moved this block to a few spots now.  Got a generic construct for us to use?